tests/unittest/cloud_governance/dynamodb_upload_data/test_upload_cloudtrail_data_to_dynamodb.py

```diff
@@ -0,0 +1,102 @@
+    default_ami_id = 'ami-03cf127a'
+    tags = [{'Key': 'Name', 'Value': 'CloudGovernanceTestInstance'}, {'Key': 'User', 'Value': 'cloud-governance'}]
+    instance_id = ec2_client.run_instances(ImageId=default_ami_id, InstanceType='t2.micro', MaxCount=1, MinCount=1, TagSpecifications=[{'ResourceType': 'instance', 'Tags': tags}])['Instances'][0].get('InstanceId')
+    ec2_client.stop_instances(InstanceIds=[instance_id])
+    dynamo_db = UploadDataToDynamoDb()
+    event_id = str(uuid.uuid1())
+    mock_log_data = [{
+        'EventId': event_id,
+        'EventTime': datetime.datetime.now(),
+        'EventName': 'StopInstances',
+        'AWS::EC2::Instance': instance_id,
+        'Username': 'cloud-governance-user-test',
+        'Resources': [{'ResourceType': 'AWS::EC2::Instance', 'ResourceName': instance_id}]
+    }]
+    dynamo_db._create_table_not_exists(primary_key='EventId')
+    dynamo_db._upload_to_dynamo_db_table(data=mock_log_data)
+    ec2_stop_obj = EC2Stop()
+    ec2_stop_obj._EC2Stop__fetch_stop_instance(sign=le, instance_days=1, delete_instance_days=0)
+    amis = ec2_client.describe_images(Owners=['self'])['Images']
+    snapshot_id = amis[0].get('BlockDeviceMappings')[0].get('Ebs').get('SnapshotId')
+    snapshots = ec2_client.describe_snapshots(OwnerIds=['self'], SnapshotIds=[snapshot_id])['Snapshots']
+    assert len(snapshots) == len(amis)
+
+
+@pytest.mark.skip(reason="May be enabled in future")
+@mock_ec2
+@mock_cloudtrail
+@mock_dynamodb2
+@mock_iam
+def test_upload_data_tag_non_cluster_resource():
+    iam_client = boto3.client('iam')
+    iam_client.create_user(UserName='cloud-governance-test', Tags=[{'Key': 'User', 'Value': 'cloud-governance-test'},
+                                                                   {'Key': 'Project', 'Value': 'cloud-governance'},
+                                                                   {'Key': 'Environment', 'Value': 'Test'}])
+    ec2_client = boto3.client('ec2', region_name=os.environ.get('AWS_DEFAULT_REGION'))
+    default_ami_id = 'ami-03cf127a'
+    mandatory_tags = {'test': 'ec2-update'}
+    instance_id = ec2_client.run_instances(ImageId=default_ami_id, InstanceType='t2.micro', MaxCount=1, MinCount=1)['Instances'][0].get('InstanceId')
+    tag_resources = TagNonClusterResources(input_tags=mandatory_tags, dry_run='no', region=os.environ.get('AWS_DEFAULT_REGION'))
+    event_id = str(uuid.uuid1())
+    mock_log_data = [{
+        'EventId': event_id,
+        'EventTime': datetime.datetime.now(),
+        'EventName': 'RunInstances',
+        'AWS::EC2::Instance': instance_id,
+        'Username': 'cloud-governance-test',
+        'Resources': [{'ResourceType': 'AWS::EC2::Instance', 'ResourceName': instance_id}]
+    }]
+    dynamo_db = UploadDataToDynamoDb()
+    dynamo_db._create_table_not_exists(primary_key='EventId')
+    dynamo_db._upload_to_dynamo_db_table(data=mock_log_data)
+    assert len(tag_resources.non_cluster_update_ec2()) == 1
+
+
```

Comment:
We dont need mock here, how its work ?

Reply:
It test data is uploaded to dynamodb table.